### DISCUSSION TITLE
How to Filter Objects by Multi-Object Custom Field in NetBox Efficiently?

### DISCUSSION THREAD

**Discussion by Brescou** [netbox-community/netbox#16906] [Help Wanted!]
Hi NetBox Team,

I'm trying to filter objects using a custom field of type multi-object. Here is the query I am using:

```python
Prefix.objects.filter(vlan=vlan, custom_field_data__my_cf_name=obj_name)
```
The issue I'm facing is that the custom field my_cf_name returns a list of IDs. If there's a single object, I get a list like [id], and if there are multiple objects, I get a list like [id1, id2, ...]. Make sense it's the  id list of the object associated. 

Is there a way to filter using this custom field without having to use multiple subqueries? I'm looking for an efficient way to match objects in the multi-object custom field directly.

Thank you!

**Comment by ghost** [comment]
@Brescou This is not a question specific to Netbox. You should check on Django' side.

So, as you found, you can filter using the custom field
https://stackoverflow.com/questions/36389871/how-to-perform-filtering-with-a-django-jsonfield

Now, you can use Django [Subquery expression](https://docs.djangoproject.com/en/5.1/ref/models/expressions/#subquery-expressions).
But in your case, your field should probably be a relational field from the start instead of storing pure ids.
